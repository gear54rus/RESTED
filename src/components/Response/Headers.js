import React, { PropTypes } from 'react';
import Highlight from 'react-highlight';

import Collapse from 'components/Collapsible';
import { responseShape } from 'propTypes/response';

function Headers({ headers, expanded }) {
  if (expanded) {
    return (
      <div>
        <h4>Headers</h4>
        <Highlight className="http">
          {headers.reduce((prev, header) => (
            `${prev ? `${prev}\n` : ''}${header.name}: ${header.value}`
          ), '')}
        </Highlight>
      </div>
    );
  }

  return (
    <Collapse
      title="Headers"
      id="Headers"
    >
      <Highlight className="http">
        {headers.reduce((prev, header) => (
          `${prev ? `${prev}\n` : ''}${header.name}: ${header.value}`
        ), '')}
      </Highlight>
    </Collapse>
  );
}

Headers.defaultProps = {
  expanded: false,
};

Headers.propTypes = {
  headers: responseShape.headers,
  expanded: PropTypes.bool,
};

export default Headers;

