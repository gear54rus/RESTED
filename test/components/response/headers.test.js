import React from 'react';
import { shallow } from 'enzyme';

/* eslint-disable import/no-unresolved */
import Headers from 'components/Response/Headers';
import Collapse from 'components/Collapsible';

describe('Headers', () => {
  describe('when passed the expanded prop', () => {
    const props = {
      headers: [{
        name: 'test',
        value: '123',
      }, {
        name: 'spasibo',
        value: 'comrade',
      }],
      expanded: true,
    };

    it('should NOT render a Collapse', () => {
      const tree = shallow(<Headers {...props} />);
      expect(tree.find(Collapse)).toBeEmpty();
    });

    it('should render a heading', () => {
      const tree = shallow(<Headers {...props} />);
      expect(tree.find('h4')).toMatchSnapshot();
    });

    it('should render the headers', () => {
      const tree = shallow(<Headers {...props} />);
      expect(tree.find('Highlight')).toMatchSnapshot();
    });
  });

  describe('when not passed the expanded prop', () => {
    const props = {
      headers: [{
        name: 'test',
        value: '123',
      }, {
        name: 'spasibo',
        value: 'comrade',
      }],
      expanded: false,
    };

    it('should render a Collapse', () => {
      const tree = shallow(<Headers {...props} />);
      expect(tree.find(Collapse)).toBePresent();
    });

    it('should render a heading', () => {
      const tree = shallow(<Headers {...props} />);
      expect(tree.find('h4')).toMatchSnapshot();
    });

    it('should render the headers', () => {
      const tree = shallow(<Headers {...props} />);
      expect(tree.find('Highlight')).toMatchSnapshot();
    });
  });
});

