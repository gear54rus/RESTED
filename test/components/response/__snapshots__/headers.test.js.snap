// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headers when not passed the expanded prop should render a heading 1`] = `null`;

exports[`Headers when not passed the expanded prop should render the headers 1`] = `
<Highlight
  className="http"
  element={null}
  innerHTML={false}
>
  test: 123
spasibo: comrade
</Highlight>
`;

exports[`Headers when passed the expanded prop should render a heading 1`] = `
<h4>
  Headers
</h4>
`;

exports[`Headers when passed the expanded prop should render the headers 1`] = `
<Highlight
  className="http"
  element={null}
  innerHTML={false}
>
  test: 123
spasibo: comrade
</Highlight>
`;
